child_compassion/models/child_pictures.py

```diff
@@ -171,12 +179,8 @@ def _get_picture(self, pic_type="Headshot", width=300, height=400):
                     self.headshot = data
                 elif pic_type.lower() == "fullshot":
                     self.fullshot = data
-            except Exception:
-                self._error_msg = (
-                    "Image cannot be fetched, invalid image "
-                    "url : " + picture.image_url
-                )
-                logger.error("Image cannot be fetched : " + picture.image_url)
-                continue
+            except Exception as e:
+                logger.error(f"Failed to fetch image from {picture.image_url}: {e}")
+                raise UserError(_(f"Failed to fetch image from URL: {picture.image_url}"))
```

Comment:
Better way for logging the full stack trace, also don't raise any error here.
```suggestion
            except (AttributeError, ValueError, URLError, HTTPError, TypeError):
                logger.error("Failed to fetch image from %s", picture.image_url, exc_info=True)
```